dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,221 @@
+                    created_on=row["ENTITY_CREATED_DATE"],
+                    user_name=row["USER_NAME"],
+                    created_by=row["CREATED_BY"],
+                    is_public=row["IS_PUBLIC"],
+                )
+            )
+        return entity_created
+
+    @task
+    def generate_slack_message(entity_created: List[EntityCreated], **context) -> str:
+        """Generate the message to be posted to the slack channel."""
+        message = ":synapse: Datasets or projects created in the last 7 days \n\n"
+        for index, row in enumerate(entity_created):
+            if row.content_type:
+                data_type = row.content_type
+            else:
+                data_type = row.node_type
+            message += f"{index+1}. <https://www.synapse.org/#!Synapse:syn{row.id}|*{row.name}*> (Type: {data_type}, Created on: {row.created_on}, Created by: <https://www.synapse.org/Profile:{row.created_by}/profile|{row.user_name}>, Public: {row.is_public})\n\n"
+        return message
+
+    @task
+    def post_slack_messages(message: str) -> bool:
+        """Post the top downloads to the slack channel."""
+        client = WebClient(token=Variable.get("SLACK_DPE_TEAM_BOT_TOKEN"))
+        result = client.chat_postMessage(channel="hotdrops", text=message)
+        print(f"Result of posting to slack: [{result}]")
+        return result is not None
+
+    @task
+    def push_results_to_synapse_table(
+        entity_created: List[EntityCreated], **context
+    ) -> None:
+        """Push the results to a Synapse table."""
+        data = []
+        # convert context["params"]["backfill_date_time"] to date in same format as date.today()
+        today = (
+            date.today()
+            if not context["params"]["backfill"]
+            else datetime.strptime(
+                context["params"]["backfill_date_time"], "%Y-%m-%d  %H:%M:%S"
+            ).date()
+        )
+        for row in entity_created:
+            data.append(
+                [
+                    row.name,
+                    row.id,
+                    row.node_type,
+                    row.parent_id,
+                    row.content_type,
+                    row.created_on,
+                    row.created_by,
+                    row.is_public,
+                    today,
+                ]
+            )
+
+        syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
+        syn_hook.client.store(
+            synapseclient.Table(schema=SYNAPSE_RESULTS_TABLE, values=data)
+        )
+
+    entity_created = get_datasets_projects_created_7_days()
+    check = check_backfill()
+    stop = stop_dag()
+    push_to_synapse_table = push_results_to_synapse_table(entity_created=entity_created)
+    slack_message = generate_slack_message(entity_created=entity_created)
+    post_to_slack = post_slack_messages(message=slack_message)
+
+    entity_created >> check >> [stop, slack_message]
+    entity_created >> slack_message
```

Comment:
I think this line (`entity_created >> slack_message`) should be removed. Since above you are already doing creating a directed edge to call `slack_message` based off the conditional present in `check`: `entity_created >> check >> [stop, slack_message]`

Reply:
Got it. Thanks